src/lib/storage/chunk.cpp

```diff
@@ -14,27 +14,29 @@
 
 namespace opossum {
 
-void Chunk::add_segment(std::shared_ptr<BaseSegment> segment) {
-  // Implementation goes here
-}
+void Chunk::add_segment(std::shared_ptr<BaseSegment> segment) { _segments.emplace_back(segment); }
 
 void Chunk::append(const std::vector<AllTypeVariant>& values) {
-  // Implementation goes here
+  DebugAssert(values.size() == _segments.size(),
+              ("Invalid row size" + std::to_string(values.size()) + " " + std::to_string(_segments.size())));
+  for (size_t column_id{0}; column_id < _segments.size(); ++column_id) {
+    _segments.at(column_id)->append(values.at(column_id));
```

Comment:
The protected access with `at()` is not necessary here because `column_id` cannot exceed `_segments.size()`. Since you have checked above that `values.size() == _segments.size()`, the second `at()` is also not necessary.